src/Faker/Provider/fa_IR/Address.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Faker\Provider\fa_IR;
+
+class Address extends \Faker\Provider\Address
+{
+    protected static $country = array(
+        'آذربایجان', 'آفریقای جنوبی', 'آفریقای مرکزی', 'آلمان', 'آنتیگوا و باربودا', 'آندورا', 'آنگولا', 'آرژانتین', 'آلبانی',
+        'اتریش', 'اتیوپی', 'اردن', 'ارمنستان', 'اریتره', 'ازبکستان', 'اسپانیا', 'استرالیا', 'استونی', 'اسلوواکی', 'اسلوونی', 'افغانستان', 'الجزایر', 'السالوادور', 'امارات متحده عربی', 'امریکا', 'اندونزی', 'اوروگوئه', 'اوگاندا', 'اوکراین', 'اکووادور', 'ایتالیا', 'ایران', 'ایرلند', 'ایسلند',
+        'باربادوس', 'باهاما', 'بحرین', 'برزیل', 'برونئی', 'بریتانیا', 'بلاروس', 'بلژیک', 'بلغارستان', 'بلیز', 'بنگلادش', 'بنین', 'بوتان', 'بوتسوانا', 'بوروندی', 'بورکینافاسو', 'بولیوی',
+        'پاپواگینه نو', 'پاراگوئه', 'پاناما', 'پاکستان', 'پرتغال', 'پرو',
+        'تاجیکستان', 'تانزانیا', 'تایلند', 'ترکمنستان', 'ترکیه', 'ترینیداد و توباگو', 'توگو', 'تونس', 'تونگا', 'تووالو', 'تایوان-چین تایپه',
+        'جامائیکا', 'جزایر سلیمان', 'جیبوتی',
+        'جمهوری جزایر مارشال', 'جمهوری چک',
+        'چاد', 'چین',
+        'دانمارک', 'دومنکن', 'دومنیکا',
+        'روآندا', 'روسیه', 'رومانی',
+        'زئیر-جمهوری کنگو', 'زامبیا', 'زلاندنو', 'زیمباوه',
+        'ژاپن',
+        'سائوتومه و پرنسیپ', 'ساحل عاج-کوت دیوار', 'ساموای غربی', 'سان مارینو', 'سریلانکا', 'سشیل', 'سنت لوسیا', 'سنت وینست و گرنادین', 'سنت کیتس و نویس', 'سنگاپور', 'سنگال', 'سوئد', 'سوئیس', 'سوازیلند', 'سودان', 'سورینام', 'سوریه', 'سومالی', 'سیرالئون',
+        'شیلی',
+        'صربستان و مونته نگرو',
+        'عراق', 'عربستان', 'عمان',
+        'غنا',
+        'فرانسه', 'فنلاند', 'فیجی', 'فیلیپین',
+        'قبرس', 'قرقیزستان', 'قزاقستان', 'قطر',
+        'کاستاریکا', 'کامبوج', 'کامرون', 'کانادا', 'کرواسی', 'کرۀ جنوبی', 'کرۀ شمالی', 'کلمبیا', 'کنیا', 'کوبا', 'کویت', 'کیپ ورد', 'کیریباس',
+        'گابون', 'گامبیا', 'گراندا', 'گرجستان', 'گواتمالا', 'گویان', 'گینه', 'گینۀ استوائی', 'گینۀ بیسائو',
+        'لائوس', 'لاتوی', 'لبنان', 'لسوتو', 'لهستان', 'لوکزامبورک', 'لیبریا', 'لیبی', 'لیتوانی', 'لیختن اشتاین',
+        'ماداگاسکار', 'مالاوی', 'مالت', 'مالدیو', 'مالزی', 'مالی', 'مجارستان', 'مصر', 'مغرب- مراکش', 'مغولستان', 'مقدونیه', 'موریتانی', 'موریس', 'موزامبیک', 'مولدووا', 'موناکو', 'مکزیک', 'میانمار(برمه)', 'میکرونوزی',
+        'نائورو', 'نامبیا', 'نپال', 'نروژ', 'نیجر', 'نیجریه', 'نیکاراگوئه',
+        'واتیکان', 'وانواتو', 'ونزوئلا', 'ویتنام',
+        'هائیتی', 'هلند', 'هندوراس', 'هندوستان',
+        'یمن', 'یونان'
+    );
+
+    protected static $state = array(
+        'آذربایجان شرقی', 'آذربایجان غربی', 'اردبیل', 'اصفهان', 'البرز', 'ایلام', 'بوشهر', 'تهران', 'چهارمحال بختیاری', 'خراسان جنوبی', 'خراسان رضوی', 'خراسان شمالی', 'خوزستان', 'زنجان', 'سمنان', 'سیستان و بلوچستان', 'فارس', 'قزوین', 'قم', 'کردستان', 'کرمان', 'کرمانشاه', 'کهکیلویه و بویراحمد', 'گلستان', 'گیلان', 'لرستان', 'مازندران', 'مرکزی', 'هرمزگان', 'همدان', 'یزد'
+    );
+
+    protected static $city = array(
```

Comment:
What's the datasource? Please read https://github.com/fzaninotto/Faker/blob/master/CONTRIBUTING.md